### DISCUSSION TITLE
Physics bounciness and degrees of freedom not working?

### DISCUSSION THREAD

**Discussion by KokoDoko** [excaliburjs/Excalibur#2365] [Q&A] [answered]
I am building a excalibur game with realistic physics, in my game I enable physics with

```
class Game {
  constructor(){
    Physics.useRealisticPhysics()
    Physics.gravity = new Vector(0, 800)
    this.engine = new Engine()
    // ...
  }
}
```
This works, and I can add Actors with physics properties, except for these two:

```
this.body.bounciness //and friction, mass, inertia
this.body.limitDegreeOfFreedom.push(DegreeOfFreedom.Rotation)
```

The body does not bounce, and can still rotate around the z axis. Am I missing something?
The whole class:

```
export class Mario extends Actor {

    constructor(texture) {
        super({ width: texture.width, height: texture.height }) // collision box size is working
        this.graphics.use(texture.toSprite())
        // enable physics
        this.body.useGravity = true // working
        this.body.collisionType = CollisionType.Active // working
        this.body.bounciness = 0.7 // not working
        this.body.limitDegreeOfFreedom.push(DegreeOfFreedom.Rotation) // not working

        // all the console logs are working
        console.log(this.body.mass)
        console.log(this.body.inertia)
        console.log(this.body.bounciness)
        console.log(this.body.friction)
        console.log(this.body)
   }
 }
```
BTW I am using the .esm version, I import excalibur as a native module from https://esm.sh/excalibur

**Comment by eonarheim** [comment, marked as answer]
@KokoDoko Thanks for pointing these out, I think I know what's up with the degree of freedom. The bounciness is a mystery still. 

I've made an issue https://github.com/excaliburjs/Excalibur/issues/2368

**Reply by eonarheim** [reply]
Okay I've got it figured out I had some incorrect math in the velocity constraint effectively making bounciness irrelevant. 👍 I should have a patch soon

**Reply by KokoDoko** [reply]
That's great, thanks!
